tracing/src/test/java/com/palantir/tracing/TracersTest.java

```diff
@@ -907,6 +907,28 @@ public void testAddTracingHeaders_populates() {
                         "X-B3-TraceId", "defaultTraceId",
                         "X-B3-Sampled", "1"));
         assertThat(headers).containsKey("X-B3-SpanId");
+        // no requestId set when span type is local
+        assertThat(headers).doesNotContainKey("Parent-Request-Id");
+    }
+
+    @Test
+    public void testAddTracingHeaders_addsParentRequestId() {
+        // need to set span type to SERVER_INCOMING to generate a requestId
+        Tracer.initTraceWithSpan(
+                Observability.SAMPLE,
+                "defaultTraceId",
+                Optional.of("forUserAgent"),
+                "rootOperation",
+                SpanType.SERVER_INCOMING);
```

Comment:
We can capture the requestId value here, and assert equality below rather than `isNotBlank()`